### ISSUE TITLE
Core: Method class needs clean up

### ISSUE THREAD

**Issue by smavros** [spirit-code/spirit#396]
Functions like
```c++
virtual std::string SolverFullName();
virtual std::string SolverName();
```
and attributes like
```c++
std::vector<std::shared_ptr<vectorfield>> configurations;
std::vector<std::shared_ptr<vectorfield>> configurations_predictor;
std::vector<std::shared_ptr<vectorfield>> configurations_temp;
```
might have to be moved to Method_Solver

**Comment by GPMueller** [comment]
Moved some things with 88c49f5a31508ecb6f53d34c05d7bec383b46e29 on `feature-improved-api`. Moving the functions will make some things cumbersome due to the fact that `Method_Solver` is templated - it's not worth the effort.

**Comment by GPMueller** [comment]
Merged into `develop` with eb841c5bd39941d4c437895845969c00ff224f10.
